src/DGtal/topology/CubicalComplex.ih

```diff
@@ -62,7 +62,7 @@ CubicalComplex()
 template <typename TKSpace, typename TCellContainer>
 inline
 DGtal::CubicalComplex<TKSpace, TCellContainer>::
-CubicalComplex( ConstAlias<KSpace> aK )
+CubicalComplex( Clone<KSpace> aK )
```

Comment:
Beware that `Clone` will allocate new `KSpace` (in many cases, e.g. when passing by const-reference) when `operator&` is applied below. Thus, it is now needed to free this memory in the destructor of `CubicalComplex`!

BTW, why changing the passing type?

Reply:
In addition, if we actually need to copy the Khalimsky space, then we should store it by value instead by pointer.

Reply:
I would love to have a copy.
I suffered the same problem than other classes had with `ConstAlias` (sorry I cannot find what class/commit changed from ConstAlias because the same problem) they make it impossible to construct the object in place (like emplacing_back to a vector).

Reply:
Do you have an example that fails ?

This works on my side:
```C++
using KSpace = DGtal::KhalimskySpaceND<3>;
using Point = KSpace::Point;
using CC = DGtal::CubicalComplex<KSpace>;

KSpace K();
K.init(Point(0, 0, 0), Point(512, 512, 512));

std::vector<C> v;
v.reserve(10); // Not actually needed
v.emplace_back(K);
```

but:
```C++
v.resize(10);
```
fails because of the lack of default constructor.

Reply:
BTW, we could also store Khalimsky space by copy in CubicalComplex but I am wondering if it was designed that way in the case of a more heavier implementation of a Cellular Grid, like the `CodedKhalimskySpaceND` from https://dgtal-team.github.io/doc-nightly/moduleCellularTopology.html#dgtal_ctopo_sec3 (@JacquesOlivierLachaud ?)

Reply:
Just as a future reference. This is the commit (and comment) changing Alias to CountedPtr in Domains.

```diff
2013-10-31 00c92bf0b Update doc and concept accordingly. Update DigitalSetFromMap. (Jacques-Olivier Lachaud)

diff --git a/src/DGtal/kernel/sets/DigitalSetBySTLVector.h b/src/DGtal/kernel/sets/DigitalSetBySTLVector.h
--- a/src/DGtal/kernel/sets/DigitalSetBySTLVector.h
+++ b/src/DGtal/kernel/sets/DigitalSetBySTLVector.h
@@ -64,9 +74,4 @@
-   * Description of template class 'DigitalSetBySTLVector' <p> \brief
-   * Aim: Realizes the concept CDigitalSet by using the STL container
-   * std::vector.
-   *
-   * It thus describes a modifiable set of points within the given
-   * domain [Domain].
-   *
-   * @tparam TDomain a realization of the concept CDomain.
-   * @see CDigitalSet,CDomain
+    @since 0.7 Domains are now hold with counted pointers and no more
+    only aliased. The problem was related to returning sets with a
+    locally constructed domain. With CountedPtr, you are sure that the
+    domain remains valid during the lifetime of your set.

```